### PR TITLE
MM-10083: document slash commands

### PR DESCRIPTION
**Pull request by lieut-data**
This adds a section under integrations re: slash commands.

### CODE REVIEW

File: site/content/integrate/slash-commands/_index.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+---
+title: "Slash Commands"
+date: "2017-08-19T12:01:23-04:00"
+section: "integrate"
+---
+
+# Slash Commands
+
+Slash commands are messages that begin with `/` and trigger an HTTP request to a web service that can in turn can post one or more messages in response.
```
Comment (jasonblais) [inline site/content/integrate/slash-commands/_index.md]: `...that can in turn post..`

File: site/content/integrate/slash-commands/_index.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+title: "Slash Commands"
+date: "2017-08-19T12:01:23-04:00"
+section: "integrate"
+---
+
+# Slash Commands
+
+Slash commands are messages that begin with `/` and trigger an HTTP request to a web service that can in turn can post one or more messages in response. 
+
+These custom slash commands are distinguished from the built-in slash commands included with Mattermost, and unlike outgoing webhooks, work in private channels and direct messages, and can be configured to auto-complete when typing.
```
Comment (jasonblais) [inline site/content/integrate/slash-commands/_index.md]: distinguished how?

File: site/content/integrate/slash-commands/_index.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+### Basic Usage
+
+Follow [the admin guide](https://docs.mattermost.com/developer/slash-commands.html#custom-slash-command) to create the custom slash command.
```
Comment (jasonblais) [inline site/content/integrate/slash-commands/_index.md]: `...to create a custom...`

File: site/content/integrate/slash-commands/_index.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+team_id=carya1qs77bemjup96ff538snh&
+text=asd&
+token=okwexkjpe7ygb8eq1ww58t483w&
+user_id=aoa1agao6t8fmx3ikt1j9w5ybw&
+user_name=somename
+```
+
+As evidenced by the `Content-Type`, this is a typical `form-urlencoded` payload similar to what you would receive from a `<form method="POST">` element on an HTML web page.
+
+Upon receiving one of these requests, your integration should immediately confirm that the `token` provided in the request body matches the one give to you when first creating the slash command. If it does not match, it is strongly recommended that you reject the request. This ensures that the request actually came from Mattermost.
```
Comment (jasonblais) [inline site/content/integrate/slash-commands/_index.md]: `...matches the one given to you after the slash command is created.`

Comment (jasonblais) [inline site/content/integrate/slash-commands/_index.md]: Is there a recommended way to confirm the `token`, or is this obvious to devs?

File: site/content/integrate/slash-commands/_index.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+Emit a message in the channel where the slash command was invoked by responding with a 200 status code and including a response body as follows:
```
Comment (jasonblais) [inline site/content/integrate/slash-commands/_index.md]: It's not clear how this follows from the previous paragraph.

File: site/content/integrate/slash-commands/_index.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+        "test_data": {
+            "ios": 78,
+            "server": 948,
+            "web": 123
+        }
+    },
+}
+```
+
+### FAQ
```
Comment (jasonblais) [inline site/content/integrate/slash-commands/_index.md]: Are these worthwhile to include, or are they admin errors? https://docs.mattermost.com/developer/slash-commands.html#troubleshooting

File: site/content/integrate/outgoing-webhooks/_index.md [hunk 7]
```diff
@@ -94,7 +94,7 @@ Outgoing webhooks support more than just the `text` field. Here is a full list o
 | username | Overrides the username the message posts as.<br> Defaults to the username set during webhook creation or the webhook creator's username if the former was not set.<br> Must be enabled [in the configuration](https://docs.mattermost.com/administration/config-settings.html#enable-integrations-to-override-usernames). | No |
 | icon\_url | Overrides the profile picture the message posts with.<br> Defaults to the URL set during webhook creation or the webhook creator's profile picture if the former was not set.<br> Must be enabled [in the configuration](https://docs.mattermost.com/administration/config-settings.html#enable-integrations-to-override-profile-picture-icons). | No |
 | attachments | [Message attachments](https://docs.mattermost.com/developer/message-attachments.html) used for richer formatting options. | If `text` is not set, yes |
-| type | Sets the post `type`, mainly for use by plugins.<br> If not blank, must begin with "custom\_". | No |
+| type | Sets the post `type`, mainly for use by plugins.<br> If not blank, must begin with "custom\_". Passing `attachments` will ignore this field and set the type to `slack\_attachment`. | No |
```
Comment (jwilander) [inline site/content/integrate/outgoing-webhooks/_index.md]: Nice add 👍

File: site/content/integrate/slash-commands/_index.md (hunk 6 above)
Comment (lieut-data) [reply site/content/integrate/slash-commands/_index.md]: Neat! I'll integrate some of those comments into this document.

**Comment by lieut-data** [comment]
Thanks for the feedback, @jasonblais! Let me know if you approve of the updated changes.

**Pull request merged**